Godeps/_workspace/src/github.com/docker/spdystream/connection.go

```diff
@@ -328,6 +337,7 @@ func (s *Connection) Serve(newHandler StreamHandler) {
 		frameQueues[partition].Push(readFrame, priority)
 	}
 	close(s.closeChan)
+	wg.Wait()
```

Comment:
> are there error conditions (non-EOF ones) that could block this wait forever?

The only time I've ever seen spdystream hang is when 1 side doesn't issue a Close() for a stream. For example, if the client creates a stream and then just calls conn.Close() (but not stream.Close()), the conn Close will wait until all outstanding streams are closed by both peers. The server could issue stream.Close(), but until the client does it too, neither side can get rid of the stream. This is why in httpstream I have its conn.Close() make sure to call Close() on all the streams. So as long as both sides defer conn.Close(), it'll all shut down cleanly.